akka-actor-tests/src/test/java/akka/pattern/PatternsTest.java

```diff
@@ -269,6 +273,38 @@ public void testCSRetry() throws Exception {
     assertEquals(expected, actual);
   }
 
+  @Test
+  public void testCSRetryWithOptional() throws Exception {
+    Callable<CompletionStage<String>> attempt =
+        () -> {
+          final CompletableFuture<String> completableFuture = new CompletableFuture<>();
+          completableFuture.completeExceptionally(new Exception("by purpose"));
+          return completableFuture;
+        };
+    final long start = System.currentTimeMillis();
+    CompletionStage<String> retriedStage =
+        Patterns.retry(
+            attempt,
+            attempted -> {
+              if (attempted == 3) {
+                return Optional.empty();
+              } else {
+                final Duration nextDuration =
+                    Duration.ofMillis(100).multipliedBy(1 + attempted.longValue());
+                return Optional.of(nextDuration);
+              }
+            },
+            system.scheduler(),
+            ec);
+    try {
+      retriedStage.toCompletableFuture().join();
+    } catch (Exception e) {
+      assertTrue(retriedStage.toCompletableFuture().isCompletedExceptionally());
+    }
+    final long elapsed = System.currentTimeMillis() - start;
+    assertTrue(elapsed > 600 && elapsed < 700);
```

Comment:
Will this test be sufficient?

Reply:
Unfortunately that is too short time margin for our CI servers. Would need something like a second interval to be stable there.

Reply:
yeah, forgot about that